src/client.rs

```diff
@@ -47,6 +47,19 @@ impl Client {
         Ok(self.connection.send_transaction(transaction).await?)
     }
 
+    pub async fn sign_transaction(&self, transaction: Transaction) -> Result<Bytes, CallError> {
+        Ok(self.connection.sign_transaction(transaction).await?)
+    }
+
+    pub async fn personal_sign(&self, data: &[&str]) -> Result<Signature, CallError> {
+        let sig = self.connection.personal_sign(data).await?.to_vec();
+        Ok(Signature {
+            r: U256::from(&sig[0..4]),
+            s: U256::from(&sig[4..8]),
+            v: sig[8] as u64,
```

Comment:
Hmm, I think this is wrong.

[`Bytes::to_vec`](https://docs.rs/ethers-core/0.6.3/ethers_core/types/struct.Bytes.html#impl) returns a `Vec<u8>`, and here the `U256` is only being created from the first 4 bytes of the vector, but 256-bit integers are 32 bytes wide and not just 4.

Reply:
https://github.com/nlordell/walletconnect-rs/commit/52c6de30c4c9a3f8fff2c5bcdfe8bb51f958c287#diff-7f93c4e263c4e9ec748f804c7fd04a3b2fde86ffd741fb5516d67e1097bae4c1R57-R61

Reply:
https://github.com/nlordell/walletconnect-rs/commit/82ab29227318b03254203d9cae62ea9ab9630d06#diff-7f93c4e263c4e9ec748f804c7fd04a3b2fde86ffd741fb5516d67e1097bae4c1R55-R56